src/algebra/group_with_zero_power.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import algebra.group_with_zero
+import algebra.group_power
+import algebra.field
+
+/-!
+# Powers of elements of groups with an adjoined zero element
+
+In this file we define integer power functions for groups with an adjoined zero element.
+This generalises the integer power function on a division ring.
+-/
+
+@[simp] lemma gwz.zero_pow {M : Type*} [monoid_with_zero M] :
+  ∀ n : ℕ, n ≠ 0 → (0 : M)^n = 0
+| 0     h := absurd rfl h
+| (k+1) h := zero_mul _
+
+namespace gwz
+variables {G : Type*} [group_with_zero G]
+
+section nat_pow
+
+@[simp, field_simps] theorem inv_pow (a : G) (n : ℕ) : (a⁻¹)^n = (a^n)⁻¹ :=
```

Comment:
```suggestion
@[simp, field_simps] theorem inv_pow (a : G) (n : ℕ) : (a⁻¹) ^ n = (a ^ n)⁻¹ :=
```
Why do people sometimes drop those spaces? Also lines 19, 32, ...

Reply:
Fixed in the whole file.